### ISSUE TITLE
[CI] JVM sigsegv somewhere during Reindex tests (likely ReindexFailureTests.testResponseOnSearchFailure)

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#20008]
I think the error is happening during `ReindexFailureTests.testResponseOnSearchFailure`

Console output: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+java9-periodic/991/console

Seems related to: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+java9-periodic/913/console

Test run errors:
```
ERROR: JVM J2 ended with an exception, command line: /var/lib/jenkins/.java/9-ea+130-openjdk/bin/java -ea -Xmx512m -Xms512m -XX:MaxDirectMemorySize=512m -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/var/lib/jenkins/workspace/elastic+elasticsearch+master+java9-periodic/modules/reindex/build/heapdump -Dtests.prefix=tests -Dtests.seed=34D4694819B67577 -Dtests.security.manager=true -Djava.io.tmpdir=./temp -Dtests.gradle=true -Dtests.artifact=reindex -Djava.awt.headless=true -Dtests.ifNoTests=fail -Dtests.task=:modules:reindex:test -Djna.nosys=true -Dtests.logger.level=WARN -Djunit4.childvm.cwd=/var/lib/jenkins/workspace/elastic+elasticsearch+master+java9-

ERROR: JVM J2 ended with an exception: Forked process returned with error code: 134. Very likely a JVM crash.  See process stdout at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+java9-periodic/modules/reindex/build/testrun/test/temp/junit4-J2-20160810_042006_208.sysout
	at com.carrotsearch.ant.tasks.junit4.JUnit4.executeSlave(JUnit4.java:1481)
	at com.carrotsearch.ant.tasks.junit4.JUnit4.access$000(JUnit4.java:121)
	at com.carrotsearch.ant.tasks.junit4.JUnit4$2.call(JUnit4.java:955)
	at com.carrotsearch.ant.tasks.junit4.JUnit4$2.call(JUnit4.java:952)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

More info from the crash log:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f062bad259d, pid=29354, tid=29773
#
# JRE version: Java(TM) SE Runtime Environment (9.0+131) (build 9-ea+131)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (9-ea+131, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# J 6563 C2 java.nio.DirectByteBuffer.get([BII)Ljava/nio/ByteBuffer; java.base@9-ea (135 bytes) @ 0x00007f062bad259d [0x00007f062bad24e0+0x00000000000000bd]
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /var/lib/jenkins/workspace/elastic+elasticsearch+master+java9-periodic/modules/reindex/build/testrun/test/J1/hs_err_pid29354.log
Could not load hsdis-amd64.so; library not loadable; PrintAssembly is disabled
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
```

Thread stack:

```
Current thread (0x00007f05b4901000):  JavaThread "elasticsearch[node_s1][search][T#1]" daemon [_thread_in_Java, id=29773, stack(0x00007f059fcfd000,0x00007f059fdfe000)]
Stack: [0x00007f059fcfd000,0x00007f059fdfe000],  sp=0x00007f059fdfbbd0,  free space=1018k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
J 6563 C2 java.nio.DirectByteBuffer.get([BII)Ljava/nio/ByteBuffer; java.base@9-ea (135 bytes) @ 0x00007f062bad259d [0x00007f062bad24e0+0x00000000000000bd]
J 4172 C1 org.apache.lucene.store.ByteBufferIndexInput.readBytes([BII)V (180 bytes) @ 0x00007f0624bfc494 [0x00007f0624bfc360+0x0000000000000134]
j  org.apache.lucene.util.packed.PackedReaderIterator.next(I)Lorg/apache/lucene/util/LongsRef;+190
j  org.apache.lucene.util.packed.PackedInts$ReaderIteratorImpl.next()J+2
j  org.apache.lucene.codecs.compressing.CompressingStoredFieldsReader$BlockState.doReset(I)V+521
j  org.apache.lucene.codecs.compressing.CompressingStoredFieldsReader$BlockState.reset(I)V+4
j  org.apache.lucene.codecs.compressing.CompressingStoredFieldsReader.document(I)Lorg/apache/lucene/codecs/compressing/CompressingStoredFieldsReader$SerializedDocument;+31
j  org.apache.lucene.codecs.compressing.CompressingStoredFieldsReader.visitDocument(ILorg/apache/lucene/index/StoredFieldVisitor;)V+2
j  org.apache.lucene.codecs.asserting.AssertingStoredFieldsFormat$AssertingStoredFieldsReader.visitDocument(ILorg/apache/lucene/index/StoredFieldVisitor;)V+32
j  org.apache.lucene.index.CodecReader.document(ILorg/apache/lucene/index/StoredFieldVisitor;)V+11
j  org.apache.lucene.index.FilterLeafReader.document(ILorg/apache/lucene/index/StoredFieldVisitor;)V+10
j  org.elasticsearch.search.fetch.FetchPhase.loadStoredFields(Lorg/elasticsearch/search/internal/SearchContext;Lorg/apache/lucene/index/LeafReaderContext;Lorg/elasticsearch/index/fieldvisitor/FieldsVisitor;I)V+11
j  org.elasticsearch.search.fetch.FetchPhase.createSearchHit(Lorg/elasticsearch/search/internal/SearchContext;Lorg/elasticsearch/index/fieldvisitor/FieldsVisitor;IILorg/apache/lucene/index/LeafReaderContext;)Lorg/elasticsearch/search/internal/InternalSearchHit;+7
j  org.elasticsearch.search.fetch.FetchPhase.execute(Lorg/elasticsearch/search/internal/SearchContext;)V+453
j  org.elasticsearch.search.SearchService.executeFetchPhase(Lorg/elasticsearch/search/fetch/ShardFetchRequest;)Lorg/elasticsearch/search/fetch/FetchSearchResult;+71
j  org.elasticsearch.search.action.SearchTransportService$FetchByIdTransportHandler.messageReceived(Lorg/elasticsearch/search/fetch/ShardFetchRequest;Lorg/elasticsearch/transport/TransportChannel;)V+8
j  org.elasticsearch.search.action.SearchTransportService$FetchByIdTransportHandler.messageReceived(Lorg/elasticsearch/transport/TransportRequest;Lorg/elasticsearch/transport/TransportChannel;)V+6
j  org.elasticsearch.transport.TransportRequestHandler.messageReceived(Lorg/elasticsearch/transport/TransportRequest;Lorg/elasticsearch/transport/TransportChannel;Lorg/elasticsearch/tasks/Task;)V+3
J 5561 C1 org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(Lorg/elasticsearch/transport/TransportRequest;Lorg/elasticsearch/transport/TransportChannel;)V (104 bytes) @ 0x00007f06246620a4 [0x00007f0624661ea0+0x0000000000000204]
j  org.elasticsearch.transport.TransportService$5.doRun()V+12
J 6140 C1 org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun()V (143 bytes) @ 0x00007f0624d1426c [0x00007f0624d14040+0x000000000000022c]
J 5225 C1 org.elasticsearch.common.util.concurrent.AbstractRunnable.run()V (32 bytes) @ 0x00007f062449e1a4 [0x00007f062449e0a0+0x0000000000000104]
j  java.util.concurrent.ThreadPoolExecutor.runWorker(Ljava/util/concurrent/ThreadPoolExecutor$Worker;)V+95 java.base@9-ea
j  java.util.concurrent.ThreadPoolExecutor$Worker.run()V+5 java.base@9-ea
j  java.lang.Thread.run()V+11 java.base@9-ea
v  ~StubRoutines::call_stub
V  [libjvm.so+0x82e100]  JavaCalls::call_helper(JavaValue*, methodHandle const&, JavaCallArguments*, Thread*)+0x3f0
V  [libjvm.so+0x82c860]  JavaCalls::call_virtual(JavaValue*, Handle, KlassHandle, Symbol*, Symbol*, Thread*)+0x1e0
V  [libjvm.so+0x8bdb0a]  thread_entry(JavaThread*, Thread*)+0x8a
V  [libjvm.so+0xcbe374]  JavaThread::thread_main_inner()+0xc4
V  [libjvm.so+0xcbe482]  JavaThread::run()+0xf2
V  [libjvm.so+0xb35972]  thread_native_entry(Thread*)+0xf2
C  [libpthread.so.0+0x7aa1]  start_thread+0xd1
```

Java Threads:
```
Java Threads: ( => current thread )
  0x00007f05b4903800 JavaThread "elasticsearch[node_s1][search][T#4]" daemon [_thread_blocked, id=29793, stack(0x00007f059f4f7000,0x00007f059f5f8000)]
  0x00007f05a4150800 JavaThread "elasticsearch[node_s1][search][T#3]" daemon [_thread_blocked, id=29784, stack(0x00007f059f7f8000,0x00007f059f8f9000)]
  0x00000000013ab800 JavaThread "elasticsearch[node_s0][search][T#4]" daemon [_thread_blocked, id=29780, stack(0x00007f059f8f9000,0x00007f059f9fa000)]
  0x00007f05a40b9000 JavaThread "elasticsearch[node_s0][search][T#3]" daemon [_thread_blocked, id=29779, stack(0x00007f059f9fa000,0x00007f059fafb000)]
  0x00000000013a9800 JavaThread "elasticsearch[node_s0][search][T#2]" daemon [_thread_blocked, id=29777, stack(0x00007f059fafb000,0x00007f059fbfc000)]
  0x00007f05b4902000 JavaThread "elasticsearch[node_s1][search][T#2]" daemon [_thread_blocked, id=29774, stack(0x00007f059fbfc000,0x00007f059fcfd000)]
=>0x00007f05b4901000 JavaThread "elasticsearch[node_s1][search][T#1]" daemon [_thread_in_Java, id=29773, stack(0x00007f059fcfd000,0x00007f059fdfe000)]
  0x00000000013c4000 JavaThread "elasticsearch[node_s0][search][T#1]" daemon [_thread_blocked, id=29772, stack(0x00007f059fdfe000,0x00007f059feff000)]
  0x00007f05a420b000 JavaThread "elasticsearch[node_s1][generic][T#9]" daemon [_thread_blocked, id=29709, stack(0x00007f059feff000,0x00007f05a0000000)]
  0x00007f05a4203800 JavaThread "elasticsearch[node_s1][generic][T#8]" daemon [_thread_blocked, id=29708, stack(0x00007f05dc054000,0x00007f05dc155000)]
  0x00007f05a4206000 JavaThread "elasticsearch[node_s0][generic][T#10]" daemon [_thread_blocked, id=29707, stack(0x00007f05dc155000,0x00007f05dc256000)]
  0x0000000001453000 JavaThread "elasticsearch[node_s0][generic][T#9]" daemon [_thread_blocked, id=29706, stack(0x00007f05dc256000,0x00007f05dc357000)]
  0x0000000001450800 JavaThread "elasticsearch[node_s0][generic][T#8]" daemon [_thread_blocked, id=29705, stack(0x00007f05dc357000,0x00007f05dc458000)]
  0x00007f05a41fe000 JavaThread "elasticsearch[node_s0][generic][T#7]" daemon [_thread_blocked, id=29704, stack(0x00007f05dc458000,0x00007f05dc559000)]
  0x00007f05a41fc800 JavaThread "elasticsearch[node_s0][index][T#2]" daemon [_thread_blocked, id=29703, stack(0x00007f05dc559000,0x00007f05dc65a000)]
  0x00007f05a41fa800 JavaThread "elasticsearch[node_s1][generic][T#7]" daemon [_thread_blocked, id=29702, stack(0x00007f05dc65a000,0x00007f05dc75b000)]
  0x000000000144e000 JavaThread "elasticsearch[node_s0][generic][T#6]" daemon [_thread_blocked, id=29701, stack(0x00007f05dc75b000,0x00007f05dc85c000)]
  0x00007f05a41f8800 JavaThread "elasticsearch[node_s0][generic][T#5]" daemon [_thread_blocked, id=29700, stack(0x00007f05dc85c000,0x00007f05dc95d000)]
  0x00007f05a41f5800 JavaThread "elasticsearch[node_s1][generic][T#6]" daemon [_thread_blocked, id=29699, stack(0x00007f05dc95d000,0x00007f05dca5e000)]
  0x000000000144b800 JavaThread "elasticsearch[node_s1][generic][T#5]" daemon [_thread_blocked, id=29698, stack(0x00007f05dca5e000,0x00007f05dcb5f000)]
  0x000000000143d800 JavaThread "elasticsearch[node_s1][warmer][T#1]" daemon [_thread_blocked, id=29696, stack(0x00007f05dcd5f000,0x00007f05dce60000)]
  0x00007f05a41e9000 JavaThread "elasticsearch[node_s0][warmer][T#1]" daemon [_thread_blocked, id=29695, stack(0x00007f05dce60000,0x00007f05dcf61000)]
  0x0000000001425800 JavaThread "elasticsearch[node_s0][refresh][T#1]" daemon [_thread_blocked, id=29691, stack(0x00007f05dcf61000,0x00007f05dd062000)]
  0x0000000001418000 JavaThread "elasticsearch[node_s1][refresh][T#1]" daemon [_thread_blocked, id=29687, stack(0x00007f05dd062000,0x00007f05dd163000)]
  0x00007f05a41c2800 JavaThread "elasticsearch[node_s0][index][T#1]" daemon [_thread_blocked, id=29685, stack(0x00007f05dd163000,0x00007f05dd264000)]
  0x00000000013ce800 JavaThread "elasticsearch[node_s1][index][T#2]" daemon [_thread_blocked, id=29682, stack(0x00007f05dd264000,0x00007f05dd365000)]
  0x0000000001417000 JavaThread "elasticsearch[node_s1][index][T#1]" daemon [_thread_blocked, id=29681, stack(0x00007f05dd365000,0x00007f05dd466000)]
  0x00007f05a4194800 JavaThread "elasticsearch[node_s0][generic][T#4]" daemon [_thread_blocked, id=29678, stack(0x00007f05dd666000,0x00007f05dd767000)]
  0x00000000013d0800 JavaThread "elasticsearch[node_s1][generic][T#4]" daemon [_thread_blocked, id=29677, stack(0x00007f05dd767000,0x00007f05dd868000)]
  0x00000000013ca000 JavaThread "elasticsearch[node_s1][generic][T#3]" daemon [_thread_blocked, id=29676, stack(0x00007f05dd868000,0x00007f05dd969000)]
  0x00000000013bb800 JavaThread "elasticsearch[node_s1][generic][T#2]" daemon [_thread_blocked, id=29675, stack(0x00007f05dd969000,0x00007f05dda6a000)]
  0x000000000135d000 JavaThread "elasticsearch[node_s0][local_transport][T#2]" daemon [_thread_blocked, id=29668, stack(0x00007f05ddc6a000,0x00007f05ddd6b000)]
  0x00007f05a4021000 JavaThread "elasticsearch[node_s1][local_transport][T#2]" daemon [_thread_blocked, id=29666, stack(0x00007f05ddd6b000,0x00007f05dde6c000)]
  0x00007f05a401f800 JavaThread "elasticsearch[node_s1][management][T#1]" daemon [_thread_blocked, id=29665, stack(0x00007f05dde6c000,0x00007f05ddf6d000)]
  0x00007f05b4927800 JavaThread "elasticsearch[node_s0][local_transport][T#1]" daemon [_thread_blocked, id=29664, stack(0x00007f05ddf6d000,0x00007f05de06e000)]
  0x0000000001351000 JavaThread "elasticsearch[node_s0][management][T#2]" daemon [_thread_blocked, id=29662, stack(0x00007f05de06e000,0x00007f05de16f000)]
  0x0000000001355000 JavaThread "elasticsearch[node_s1][local_transport][T#1]" daemon [_thread_blocked, id=29661, stack(0x00007f05e809e000,0x00007f05e819f000)]
  0x0000000001336000 JavaThread "elasticsearch[node_s1][generic][T#1]" daemon [_thread_blocked, id=29659, stack(0x00007f05e819f000,0x00007f05e82a0000)]
  0x00007f05a401b000 JavaThread "elasticsearch[node_s0][generic][T#3]" daemon [_thread_blocked, id=29657, stack(0x00007f05e82a0000,0x00007f05e83a1000)]
  0x00007f05b490a800 JavaThread "elasticsearch[node_s1][clusterService#updateTask][T#1]" daemon [_thread_blocked, id=29656, stack(0x00007f05e83a1000,0x00007f05e84a2000)]
  0x00007f05b4909000 JavaThread "elasticsearch[node_s1][[ttl_expire]]" daemon [_thread_blocked, id=29655, stack(0x00007f05e84a2000,0x00007f05e85a3000)]
  0x00007f05b48ff000 JavaThread "elasticsearch[node_s1][scheduler][T#1]" daemon [_thread_blocked, id=29654, stack(0x00007f05e85a3000,0x00007f05e86a4000)]
  0x00007f05a4010800 JavaThread "elasticsearch[node_s0][generic][T#2]" daemon [_thread_blocked, id=29653, stack(0x00007f05e86a4000,0x00007f05e87a5000)]
  0x00007f05b48fc800 JavaThread "elasticsearch[node_s1][[timer]]" daemon [_thread_blocked, id=29652, stack(0x00007f05e87a5000,0x00007f05e88a6000)]
  0x00007f05a4009000 JavaThread "elasticsearch[node_s0][generic][T#1]" daemon [_thread_blocked, id=29651, stack(0x00007f05ea1ab000,0x00007f05ea2ac000)]
  0x00007f05a4004000 JavaThread "elasticsearch[node_s0][management][T#1]" daemon [_thread_blocked, id=29650, stack(0x00007f05e9da7000,0x00007f05e9ea8000)]
  0x00007f05b48f9800 JavaThread "elasticsearch[node_s0][clusterService#updateTask][T#1]" daemon [_thread_blocked, id=29645, stack(0x00007f05e9ca6000,0x00007f05e9da7000)]
  0x00007f05b48f5800 JavaThread "elasticsearch[node_s0][[ttl_expire]]" daemon [_thread_blocked, id=29642, stack(0x00007f05e9ea8000,0x00007f05e9fa9000)]
  0x00007f05b48ad800 JavaThread "elasticsearch[node_s0][scheduler][T#1]" daemon [_thread_blocked, id=29634, stack(0x00007f05ea0aa000,0x00007f05ea1ab000)]
  0x00007f05b4255800 JavaThread "elasticsearch[node_s0][[timer]]" daemon [_thread_blocked, id=29561, stack(0x00007f05e9fa9000,0x00007f05ea0aa000)]
  0x00007f05ac149800 JavaThread "TEST-ReindexFailureTests.testResponseOnSearchFailure-seed#[22DD247E228C1140]" [_thread_blocked, id=29559, stack(0x00007f05eaac5000,0x00007f05eabc6000)]
  0x00007f063c58e800 JavaThread "SUITE-ReindexFailureTests-seed#[22DD247E228C1140]" [_thread_blocked, id=29557, stack(0x00007f05ea4ac000,0x00007f05ea5ad000)]
  0x00007f05b40cd000 JavaThread "process reaper" daemon [_thread_blocked, id=29456, stack(0x00007f064490c000,0x00007f0644946000)]
  0x00007f063c40e000 JavaThread "JUnit4-serializer-daemon" daemon [_thread_blocked, id=29444, stack(0x00007f05eb1cf000,0x00007f05eb2d0000)]
  0x00007f063c333000 JavaThread "Service Thread" daemon [_thread_blocked, id=29437, stack(0x00007f05eb3d1000,0x00007f05eb4d2000)]
  0x00007f063c2a0800 JavaThread "Common-Cleaner" daemon [_thread_blocked, id=29430, stack(0x00007f05eb4d2000,0x00007f05eb5d3000)]
  0x00007f063c285000 JavaThread "Sweeper thread" daemon [_thread_blocked, id=29427, stack(0x00007f05eb9fa000,0x00007f05ebafb000)]
  0x00007f063c283000 JavaThread "C1 CompilerThread2" daemon [_thread_blocked, id=29425, stack(0x00007f05ebafb000,0x00007f05ebbfc000)]
  0x00007f063c281000 JavaThread "C2 CompilerThread1" daemon [_thread_in_native, id=29424, stack(0x00007f05ebbfc000,0x00007f05ebcfd000)]
  0x00007f063c27f000 JavaThread "C2 CompilerThread0" daemon [_thread_in_native, id=29423, stack(0x00007f05ebcfd000,0x00007f05ebdfe000)]
  0x00007f063c27c800 JavaThread "Signal Dispatcher" daemon [_thread_blocked, id=29422, stack(0x00007f05ebdfe000,0x00007f05ebeff000)]
  0x00007f063c27a800 JavaThread "Reference Pending List Locker" daemon [_thread_blocked, id=29421, stack(0x00007f05ebeff000,0x00007f05ec000000)]
  0x00007f063c255000 JavaThread "Finalizer" daemon [_thread_blocked, id=29411, stack(0x00007f062002b000,0x00007f062012c000)]
  0x00007f063c251000 JavaThread "Reference Handler" daemon [_thread_blocked, id=29410, stack(0x00007f062012c000,0x00007f062022d000)]
  0x00007f063c012800 JavaThread "main" [_thread_blocked, id=29360, stack(0x00007f06449ab000,0x00007f0644aac000)]
Other Threads:
  0x00007f063c245000 VMThread "VM Thread" [stack: 0x00007f062022d000,0x00007f062032e000] [id=29406]
  0x00007f063c336000 WatcherThread [stack: 0x00007f05eb2d0000,0x00007f05eb3d1000] [id=29438]
  0x00007f063c02a800 GCTaskThread "GC Thread#0" [stack: 0x00007f0641598000,0x00007f0641699000] [id=29369]
  0x00007f063c02c000 GCTaskThread "GC Thread#1" [stack: 0x00007f0641497000,0x00007f0641598000] [id=29370]
  0x00007f063c02d800 GCTaskThread "GC Thread#2" [stack: 0x00007f0641396000,0x00007f0641497000] [id=29371]
  0x00007f063c02f800 GCTaskThread "GC Thread#3" [stack: 0x00007f0641295000,0x00007f0641396000] [id=29372]
  0x00007f063c057800 ConcurrentGCThread "G1 Main Marker" [stack: 0x00007f064048b000,0x00007f064058c000] [id=29378]
  0x00007f063c059000 ConcurrentGCThread "G1 Marker#0" [stack: 0x00007f064038a000,0x00007f064048b000] [id=29379]
  0x00007f063c037000 ConcurrentGCThread "G1 Refine#0" [stack: 0x00007f0640a8d000,0x00007f0640b8e000] [id=29376]
  0x00007f063c035000 ConcurrentGCThread "G1 Refine#1" [stack: 0x00007f0640b8e000,0x00007f0640c8f000] [id=29375]
  0x00007f063c033800 ConcurrentGCThread "G1 Refine#2" [stack: 0x00007f0640c8f000,0x00007f0640d90000] [id=29374]
  0x00007f063c031800 ConcurrentGCThread "G1 Refine#3" [stack: 0x00007f0640d90000,0x00007f0640e91000] [id=29373]
  0x00007f063c039000 ConcurrentGCThread "G1 Young RemSet Sampling" [stack: 0x00007f064098c000,0x00007f0640a8d000] [id=29377]
Threads with active compile tasks:
C2 CompilerThread16312       4       org.elasticsearch.common.io.stream.StreamOutput::writeString (136 bytes)
C2 CompilerThread06584       4       java.util.TimSort::binarySort (223 bytes)
```

Assigning to @nik9000 since it may be reindex related?  Feel free to pass the buck onto someone else, wasn't really sure who was best to handle this :)  Also /cc @mikemccand since you mentioned this on the last failure:

> This sure looks like ES is trying to use a closed IndexRedaer to load a document ... the test case seems to be ReindexFailureTests.testResponseOnSearchFailure

**Comment by nik9000** [comment]
```
J 6563 C2 java.nio.DirectByteBuffer.get([BII)Ljava/nio/ByteBuffer; java.base@9-ea (135 bytes) @ 0x00007f062bad259d [0x00007f062bad24e0+0x00000000000000bd]
```
Looks more like https://github.com/elastic/elasticsearch/issues/19861 at first glance.

**Comment by polyfractal** [comment]
Whoops, I think you're right :)

**Comment by mikemccand** [comment]
Indeed, a bad bad bug is lurking in ES ... somehow ES is attempting to use a closed IndexReader, and this test is provoking it.

This is what makes #19861 so important: we need to know when an `AlreadyClosedException` is thrown by Lucene: this is its desperate (and only best effort) attempt to avoid a SIGSEGV.

**Comment by mikemccand** [comment]
We need to fix this bug before release 5.0.0!  Risking SIGSEGV is no good.

**Comment by s1monw** [comment]
I think this is an issue in SearchContext - We use the SearchContext to hold on to an IndexSearcher and once the context goes out of scope we close it. Now, we never really protect against concurrent use of the searcher and the context. There might be a concurrent closing thread due to an exception or something like this that closes the context while somebody is using it? There is I think even a small window that allows the reaper thread to close an idle searcher while the user is coming back. like:

 - user fetches the context 

 - user updates the timestamp while the reaper has read it already
 - user calls a method on IR while reaper closes it
 - lucene hits an SIGSEGV on the low level.

I think we have to incRef the searcher / reader / context before we use it here otherwise this can always happen.
